src/forms/datepicker/DatePicker.tsx

```diff
@@ -0,0 +1,423 @@
+        if (typeof date === "string") return date;
+        return Moment(date)
+            .locale(locale)
+            .format("L");
+    }
+
+    private handleInputChange = (evt: React.ChangeEvent<HTMLInputElement>) => {
+        this.setState(
+            {
+                value: evt.target.value,
+                navValue: Moment(
+                    DatePicker.validDate(new Date(evt.target.value)),
+                    Moment.localeData(this.props.locale).longDateFormat("L"),
+                ),
+            },
+            this.afterInputChange,
+        );
+    };
+
+    afterInputChange = () => {
+        if (this.props.onChange) this.props.onChange(this.state.value);
+    };
+
+    render() {
+        const {isOpen, viewMode, navValue, inputFocused, value} = this.state;
+        const {locale} = this.props;
+        const navMonth = navValue.month();
+        const navYear = navValue.year();
+        const daysFromPrevMonth = DatePicker.numDaysFromPrevMonth(navYear, navMonth);
+        let calendar = Moment()
+            .year(navYear)
+            .month(navMonth)
+            .date(1)
+            .subtract(daysFromPrevMonth, "days");
+        const moment = Moment(value, Moment.localeData(locale).longDateFormat("L"));
+        const selectedSameMonth = moment.isSame(navValue, "month");
+        return (
+            <div ref={this.calRef} className="clr-control-container">
+                <div className="clr-input-wrapper">
+                    <div
+                        className={classNames([
+                            "clr-input-group", //prettier
+                            inputFocused && "clr-focus",
+                        ])}
+                    >
+                        <input
+                            ref={this.inputRef}
+                            type="text"
+                            className="clr-input"
+                            placeholder={Moment.localeData(locale).longDateFormat("L")}
+                            value={value}
+                            onFocus={() => {
+                                this.setState({inputFocused: true});
+                            }}
+                            onBlur={() => {
+                                this.setState({inputFocused: false});
+                            }}
+                            onChange={this.handleInputChange}
+                        />
+                        <button
+                            className="clr-input-group-icon-action"
+                            type="button"
+                            title="Open"
+                            onClick={this.handleToggle.bind(this)}
+                        >
+                            <Icon shape="calendar" />
+                        </button>
+                        {isOpen && (
+                            <div
+                                className="datepicker"
+                                style={{
```

Comment:
@alexander-rassanov I'm not familiar with how to supply inline className value instead of style tag.  This is corner case to get the div working correctly.  Probably not a huge deal.  What do you think?

Reply:
probably we can omit for right now and will discuss later how we will deal with cases like this